python/dolfinx/fem/petsc.py

```diff
@@ -594,25 +594,12 @@ def apply_lifting(
             then ``a`` is a 1D sequence. If ``b`` is blocked or a nest,
             then ``a`` is  a 2D array of forms, with the ``a[i]`` forms
             used to modify the block/nest vector ``b[i]``.
-        bcs: Boundary conditions used to modify ``b`` (see
-            :func:`dolfinx.fem.apply_lifting`). Two cases are supported:
-
-            1. The boundary conditions ``bcs`` are a
-               'sequence-of-sequences' such that ``bcs[j]`` are the
-               Dirichlet boundary conditionns associated with the forms in
-               the ``j`` th colulmn of ``a``. Helper functions exist to
-               create a sequence-of-sequences of `DirichletBC` from the 2D
-               ``a`` and a flat Sequence of `DirichletBC` objects ``bcs``::
-
-                   bcs1 = fem.bcs_by_block(
-                    fem.extract_function_spaces(a, 1), bcs
-                   )
-
-            2. ``bcs`` is a sequence of :class:`dolfinx.fem.DirichletBC`
-               objects. The function deduces which `DiricletBC` objects
-               apply to each column of ``a`` by matching the
-               :class:`dolfinx.fem.FunctionSpace`.
-
+        bcs: Boundary conditions to apply. If ``b`` is nested or
```

Comment:
Well, I copied the documentation from `set_bc` as a reminder of what I am trying to achieve, but I am not there yet.

Comparison between the two functions:
- `set_bc` requires 1D `bcs` input for non-blocked problems, and 2D `bcs` for blocked problem.
- ideally, `apply_lifting` should do the same, but:
   - on `main`, it accepts 1D `bcs` input for either blocked or non-blocked problems. For non-blocked problems, it internally converts into a 2D array.
   - the goal would be to requires 1D `bcs` input for non-blocked problems, and 2D `bcs` for blocked problem. However:
   - this branch, as it is now, forces 2D `bcs` input, which is still not what I would have wanted in #3786. I'll fix this, but I'll probably wait for a discussion if #3786 is worth fixing or not.

Reply:
No, scratch this. `dolfinx.fem.apply_lifting` requires the input to be 2D, so that same requirement will go into `dolfinx.fem.petsc.apply_lifting`.